sessions/react-state-3/journal-app-filter/README.md

```diff
@@ -0,0 +1,103 @@
+
+Now you can click on the two filter buttons and change the `filter` state 🚀
+
+### The filter logic
+
+You will notice the displayed entries will not change yet. This is what we solve next.
+
+- Besides the `entries` array which is stored in state, create a new array called `favoriteEntries`, which **should not** be stored in state.
+- The value of `favoriteEntries` should be all element from the `entries` array where `isFavorite` is `true` (hint: you can use the array method `filter`)
+
+You have two arrays containing entries now: a list of all entries and a list of the favorite entries. The code has to make a decision which of them should be displayed. The component that renders the entry list accepts a prop for the list of entries that should be display. Here it's the `entries` prop of the `EntriesSection` component.
+
+- Pass the value to this prop conditionally. Depending on the value of the state `filter` you pass the array `entries` or the array `favoriteEntries` (hint: you can use the ternary operator)
+
+Now you will see different entries in the UI when clicking the filter buttons 🚀
+
+> 💡 If there are no entries displayed after you clicked on the "Favorites" button, you might need to click the "star" button of some entries first.
+
+### Visual highlight for active filter
+
+We are almost done. There are two minor things, that need to be fixed. First you might notice that one of the two filter buttons is highlighted as active. Clicking the buttons does not change this visual indicator yet.
+
+- Alongside the `entries` prop also pass down the `filter` state as prop to the component rendering the two buttons. (Here it's the `EntriesSection` component)
+- Inside of the `EntriesSection` component add the `filter` prop to the function declaration.
+
+Locate the prop that enabled the visual highlight for the buttons. In the template it's the `isActive` prop of the `Tab` and `Badge` components.
+
+- Pass the value of the `isActive` prop conditionally depending on the value of `filter`, which could be `"all"` or `"favorites"`.
+
+Click on the filter buttons now. You will see that the visual highlight gets applied to the button you clicked, while the other button looses the visual highlight 🚀
+
+### Show count of all and favorite entries
+
+The last thing we need to adjust is the number inside the filter buttons. They indicate how many entries are in each of the two lists. You might notice they show a random hardcoded number now.
+
+Stay in the component rendering the two buttons. (Here it's the `EntriesSection` component)
+
+- Add two new props to the function declaration: `allEntriesCount` and `favoriteEntriesCount`
+
+Go up in the component hierarchy to the place where this component is rendered.
+
+- Add the props `allEntriesCount` and `favoriteEntriesCount` props and pass a value for each.
+- The value can be read from the `length` property of the two arrays `entries` and `favoriteEntries`.
```

Comment:
```suggestion
- The value can be read from the `length` property of the two arrays `entries` and `favoriteEntries`.
```
Should we mention that this property can be accessed via dot notation?

Reply:
```suggestion
- The value can be read from the `length` property of the two arrays `entries` and `favoriteEntries`. You can access the `length` property by using the dot notion on the array variables.
```

Reply:
Couldn't hurt. I added a note